ggml/src/ggml-sycl/ggml-sycl.cpp

```diff
@@ -1795,10 +1789,15 @@ static  void pool2d_nchw_kernel(
         const int ew = sycl::min(iw, start_w + kw);
 
         To res = 0;
+        bool op_valid = true;
```

Comment:
Why is `op_valid` needed? Can't we just set `res` to `NaN` in the `default` cases and avoid the `if-else` block at the end? This could avoid some performance overhead.